pkg/virt-launcher/virtwrap/converter/converter.go

```
@@ -132,10 +133,17 @@ func Convert_v1_Disk_To_api_Disk(c *ConverterContext, diskDevice *v1.Disk, disk
 	if diskDevice.Disk != nil {
 		var unit int
 		disk.Device = "disk"
+		disk.Serial = diskDevice.Serial
 		disk.Target.Bus = diskDevice.Disk.Bus
 		disk.Target.Device, unit = makeDeviceName(diskDevice.Name, diskDevice.Disk.Bus, prefixMap)
 		if diskDevice.Disk.Bus == "scsi" {
 			assignDiskToSCSIController(disk, unit)
+
+			// Force truncation of serial number to MaxSCSISerialLen characters, as QEMU no longer does this automatically.
+			// This is required to maintain backward compatibility. Specifying devices with serial numbers longer than MaxSCSISerialLen
+			// characters is not allowed now.
+			disk.Serial = hwutil.TruncateSCSIDiskSerial(diskDevice.Serial)
```

I'd prefer not to silently truncate values, even if it breaks compatibility.